internal/fsutil/create.go

```diff
@@ -13,10 +13,16 @@ type CreateOptions struct {
 	Mode fs.FileMode
 	Data io.Reader
 	Link string
+	Dirs bool // create missing parent directories with 0755 mode
```

Comment:
Although it's satisfying to have another four-letter field name, I don't think the name 'dirs' makes it obvious what this field does (this is also highlighted to me by the need for a comment). Maybe something more descriptive would be clearer, perhaps `MakeParentDirs`?

Also, I think the 0755 here might be incorrect, is the mode actually `fs.ModeDir | 0775`? Perhaps instead of '0755 mode', 'permissions 0755' would be more accurate? More generally though, this comment feels to me like it's describing an action rather than a property of this field so perhaps it's worth instead mentioning something to this effect on the `Create` function?

Reply:
Per `chmod(1)`:

> This  manual  page  documents  the  GNU version of chmod.  chmod changes the file mode bits of each given file according to mode, which can be either a symbolic representation of changes to make, or an octal number representing the bit pattern for the new mode bits.

It's pretty common to refer to permission bits as mode, and I think the comment makes it obvious that this doesn't refer to file type bits (it explicitly says _directories_ and `0755` doesn't contain file type bits).

Anyway, applied the suggestions. @TheSignPainter98 please resolve this thread if appropriate.

Reply:
Ah thanks for the info, I wasn't aware of this convention. Changes look good to me

Reply:
Not to be a bikeshedding nightmare, but we have used `MakeParents` in Pebble and `ops` - maybe worth sticking with that? 

https://github.com/search?q=repo%3Acanonical%2Fpebble%20parents&type=code

Reply:
@jnsgruk updated to `MakeParents`. (I like it more than the longer variant as well.)